### DISCUSSION TITLE
Hubbard-Holstein 2-RDM Properties

### DISCUSSION THREAD

**Discussion by iavdic** [block-hczhai/block2-preview#148] [Q&A] [answered]
Hi Huanchen,

When using the get_2pdm() function with, for example, the MPS from the Hubbard-Holstein model, are the bosonic degrees of freedom traced out and the output 2-RDM is purely a fermionic one, or is the output a fermion-boson 2-RDM?

Thanks!

**Comment by hczhai** [comment, marked as answer]
Thanks for your interest in the ``block2`` package.

For the Hubbard-Holstein model, the 2PDM is not a well-defined quantity, and you will get runtime error if you call ``driver.get_2pdm`` directly for the MPS from the Hubbard-Holstein model. You may compute the individual 2PDM elements using ``driver.expectation`` instead.

**Reply by iavdic** [reply]
Okay, thanks!
